cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaMirrorMakerAssemblyOperator.java

```diff
@@ -0,0 +1,96 @@
+ * </ul>
+ */
+public class KafkaMirrorMakerAssemblyOperator extends AbstractAssemblyOperator<KubernetesClient, KafkaMirrorMaker, KafkaMirrorMakerList, DoneableKafkaMirrorMaker, Resource<KafkaMirrorMaker, DoneableKafkaMirrorMaker>> {
+
+    private static final Logger log = LogManager.getLogger(KafkaMirrorMakerAssemblyOperator.class.getName());
+
+    private final DeploymentOperator deploymentOperations;
+
+    /**
+     * @param vertx                      The Vertx instance
+     * @param isOpenShift                Whether we're running with OpenShift
+     * @param mirrorMakerOperator        For operating on MirrorMakers
+     * @param networkPolicyOperator      For operating on NetworkPolicies
+     * @param secretOperations           For operating on Secrets
+     */
+    public KafkaMirrorMakerAssemblyOperator(Vertx vertx, boolean isOpenShift,
+                                            CertManager certManager,
+                                            CrdOperator<KubernetesClient, KafkaMirrorMaker, KafkaMirrorMakerList, DoneableKafkaMirrorMaker> mirrorMakerOperator,
+                                            SecretOperator secretOperations,
+                                            NetworkPolicyOperator networkPolicyOperator,
+                                            DeploymentOperator deploymentOperations) {
+        super(vertx, isOpenShift, ResourceType.MIRRORMAKER, certManager, mirrorMakerOperator, secretOperations, networkPolicyOperator);
+        this.deploymentOperations = deploymentOperations;
+    }
+
+    @Override
+    protected Future<Void> createOrUpdate(Reconciliation reconciliation, KafkaMirrorMaker assemblyResource) {
+        String namespace = reconciliation.namespace();
+        String name = reconciliation.name();
+        KafkaMirrorMakerCluster mirror;
+        try {
+            mirror = KafkaMirrorMakerCluster.fromCrd(assemblyResource);
+        } catch (Exception e) {
+            return Future.failedFuture(e);
+        }
+
+        log.debug("{}: Updating Kafka Mirror Maker cluster", reconciliation, name, namespace);
+        return deploymentOperations.scaleDown(namespace, mirror.getName(), mirror.getReplicas())
+                //.compose(i -> configMapOperations.reconcile(namespace, mirror.getAncillaryConfigName(), logAndMetricsConfigMap))
+                .compose(i -> deploymentOperations.reconcile(namespace, mirror.getName(), mirror.generateDeployment(null)))
+                .compose(i -> deploymentOperations.scaleUp(namespace, mirror.getName(), mirror.getReplicas()).map((Void) null));
+    }
+
+    @Override
+    protected Future<Void> delete(Reconciliation reconciliation) {
+        String namespace = reconciliation.namespace();
+        String assemblyName = reconciliation.name();
+        String clusterName = KafkaMirrorMakerCluster.kafkaMirrorMakerClusterName(assemblyName);
+
+        return deploymentOperations.reconcile(namespace, clusterName, null).map((Void) null);
```

Comment:
Do we really need this? We should be using the ownerReference to delete the objects.